src/cpp/rtps/RTPSDomain.cpp

```diff
@@ -154,16 +154,19 @@ RTPSParticipant* RTPSDomain::createParticipant(
         guidP.value[11] = octet(ID >> 24);
     }
 
-    // If we force the participant to have a specific prefix we must define a different persistence GuidPrefix_t for 
-    // its endpoint. This instance-bounded persitence GuidPrefix_t will be kept in the participant instance local copy
-    // of the attributes (prefix member).
+    RTPSParticipant* p = new RTPSParticipant(nullptr);
+    RTPSParticipantImpl* pimpl = nullptr;
+
+    // If we force the participant to have a specific prefix we must define a different persistence GuidPrefix_t that
+    // would ensure builtin endpoints are able to differentiate between a communication loss and a participant recovery
     if(PParam.prefix != c_GuidPrefix_Unknown)
     {
-        std::swap(guidP, PParam.prefix);
+        pimpl = new RTPSParticipantImpl(PParam, PParam.prefix, guidP, p, listen);
+    }
+    else
+    {
+        pimpl = new RTPSParticipantImpl(PParam, guidP, p, listen);
```

Comment:
The RTPSParticipantImpl constructor overload with the new persistent guid parameter has actually replace the old one. The old constructor calls the new one passing c_GuidPrefix_Unknown. This was actually force by the  C++ standard because you cannot call from a generic constructor a more specialized one using member initialization syntax.

Why keep the old overload? To hint the user that its actually the more usual case and the persistent guid parameter is an advance feature. Avoiding this generates cryptic code in a source base that is rather cryptic already.